File: Exercícios em grupo/7_programacao-avancada-front-end/sgcmapp/src/app/service/i-service.ts
import { Observable } from "rxjs";
import { PageRequest } from "../model/page-request";
import { PageResponse } from "../model/page-response";

export interface IService<T> {

    apiUrl: string;

    get(termoBusca?: string, pageRequest?: PageRequest): Observable<PageResponse<T>>;
    getById(id: number): Observable<T>;
    save(objeto: T): Observable<T>;
    delete(id: number): Observable<void>;

}

File: Exercícios em grupo/9_topicos-emergentes/sgcmapi/src/main/java/br/ufac/sgcmapi/service/IService.java
package br.ufac.sgcmapi.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface IService<T> {

    public Page<T> get(Pageable page);
    public T get(Long id);
    public Page<T> get(String termoBusca, Pageable page);
    public T save(T objeto);
    public void delete(Long id);
    
}
